addons/product/models/product_uom.py

```diff
@@ -0,0 +1,32 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductUom(models.Model):
+    _description = 'Link between products and their UoMs'
+    _rec_name = 'barcode'
+
+    uom_id = fields.Many2one('uom.uom', 'Unit', required=True, ondelete='cascade')
+    product_id = fields.Many2one('product.product', 'Product', required=True, ondelete='cascade')
+    barcode = fields.Char(index='btree_not_null', required=True)
+    company_id = fields.Many2one('res.company', 'Company', default=lambda self: self.env.company)
```

Comment:
So you have to define the barcode for every company? Or can you define that to be cross-company as well?